src/compiler/checker.ts

```diff
@@ -4722,6 +4846,1104 @@ namespace ts {
+                // The symbol serializer ends up creating many sibling scopes that all need "separate" contexts when
+                // it comes to naming things - within a normal `typeToTypeNode` call, the node builder only ever descends
+                // through the type tree, so the only cases where we could have used distinct sibling scopes was when there
+                // were multiple generic overloads with similar generated type parameter names
+                // The effect:
+                // When we write out
+                // export const x: <T>(x: T) => T
+                // export const y: <T>(x: T) => T
+                // we write it out like that, rather than as
+                // export const x: <T>(x: T) => T
+                // export const y: <T_1>(x: T_1) => T_1
+                if (initial.typeParameterNames) {
+                    initial.typeParameterNames = cloneMap(initial.typeParameterNames);
+                }
+                if (initial.typeParameterNamesByText) {
+                    initial.typeParameterNamesByText = cloneMap(initial.typeParameterNamesByText);
+                }
+                if (initial.typeParameterSymbolList) {
+                    initial.typeParameterSymbolList = cloneMap(initial.typeParameterSymbolList);
+                }
+                return initial;
+            }
+
+            function symbolTableToDeclarationStatements(symbolTable: SymbolTable, context: NodeBuilderContext, bundled?: boolean): Statement[] {
+                // TODO: Use `setOriginalNode` on original declarations where possible so these declarations see some kind of declaration mapping
+                // We save this off here so it's not adjusted by well-meaning declaration emit codepaths which want to apply more specific contexts
+                const enclosingDeclaration = context.enclosingDeclaration!;
+                let results: Statement[] = [];
+                const visitedSymbols: Map<true> = createMap();
+                let deferredPrivates: Map<Symbol> | undefined;
+                const oldcontext = context;
+                context = {
+                    ...oldcontext,
+                    usedSymbolNames: mapMap(symbolTable, (_symbol, name) => [unescapeLeadingUnderscores(name), true]),
+                    remappedSymbolNames: createMap(),
+                    tracker: {
+                        ...oldcontext.tracker,
+                        trackSymbol: (sym, decl, meaning) => {
+                            const acessibleResult = isSymbolAccessible(sym, decl, meaning, /*computeALiases*/ false);
+                            if (acessibleResult.accessibility === SymbolAccessibility.Accessible) {
+                                // Lookup the root symbol of the chain of refs we'll use to access it and serialize it
+                                const chain = lookupSymbolChainWorker(sym, context, meaning);

```

Comment:
why don't you call `lookupSymbolChain`? Are we in the dynamic scope of some other `lookupSymbolChain` call?